airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -650,6 +652,8 @@ class _suppress(AbstractContextManager):
 
     The caught exception is also stored on the context manager instance under
     attribute ``exception``.
+
+    :meta private:
```

Comment:
Do we still need `_suppress` to be a standalone class/function, or can we simply merge it into `_optionally_suppress`?

Reply:
not _needed_ of course, but i think the separation of concerns is helpful.  we could define _suppress inside of _optionally_suppress, but... it's already tested this way so... why bother?  and it's private so it's not increasing backcompat surface area i.e. doesn't really hurt to leave it there IMO.  but you tell me

Reply:
I was thinking we could do

```python
class _optionall_suppress(AbstractContextManager):
    ...
```

since a lot of the logic in the class doesn’t really seem relevant now. But it’s OK if you feel this is easier.

Reply:
yeah so ... it's just ... one is "suppress with logging" (with otherwise identical semantics to stdlib suppress) the other is "conditionally use ... suppress with logging" which seemed to me nice from perspective of intelligibility.  rather than mashing into one where, it's entirely diff from stdlib and harder to understand what it's doing.

ALAS, i have merged them, and added a test or two